pytripgui/main_window_qt_vc/main_window_view.py

```diff
@@ -33,20 +32,27 @@ def get_kernel_config_view():
 
     def browse_file_path(self, name, extension, path=None):
         """
-        :return full file path, or empty string
+        :return full file path (absolute), or empty string
         """
-        selected_file = QFileDialog.getOpenFileName(self.ui, name, path, extension)
-        return selected_file[0]
+        selected_file_path, selected_file_extension = QFileDialog.getOpenFileName(self.ui, name, path, extension)
+        return selected_file_path
+
+    def save_file_path(self, caption, extension, path=None):
+        """
+        :return full file path (absolute), or empty string
+        """
+        selected_file_path, selected_file_extension = QFileDialog.getSaveFileName(self.ui, caption, path, extension)
+        return selected_file_path
 
     def browse_folder_path(self, name, path=None):
         """
-        :return full file path, or empty string
+        :return full directory path (absolute), or empty string
         """
         dialog = QFileDialog(self.ui, name, path)
         dialog.setFileMode(QFileDialog.Directory)
         dialog.setOptions(QFileDialog.ShowDirsOnly)
         dialog.exec_()
-        selected_path = os.path.join(dialog.directory().path(), '')
+        selected_path = dialog.selectedFiles()[0]
```

Comment:
Explain why `[0]` needs to be used here

Reply:
Comment added. The only method that extracts selected files from a file dialog window returns a list, even though this window only allows you to select a single file.
https://github.com/pytrip/pytripgui/blob/284f696219f254469a8a8ea5b5407463d711d657/pytripgui/main_window_qt_vc/main_window_view.py#L62